### PR TITLE
Implement receive parcel use case

### PR DESCRIPTION
**Pull request by edsinkevic**
The IT test is getting too big, I'll refactor it in my next PR

### CODE REVIEW

File: Colorblind/Domain/Values/ParcelStatus.cs [hunk 1]
```diff
@@ -6,5 +6,6 @@ public enum ParcelStatus
     Submitted = 8,
     Shipped = 16,
     Delivered = 32,
-    Unregistered = 64
+    Unregistered = 64,
+    Received = 128
```
Comment (SimonasKalpokas1) [inline Colorblind/Domain/Values/ParcelStatus.cs:10]: You could us [Flags] attribute for this enumeration, also you are missing 2 and 4. Could you explain when would it be possible to have multiple ParcelStatuses on one parcel?

Comment (edsinkevic) [reply Colorblind/Domain/Values/ParcelStatus.cs:10]: I didn't think too hard when I was writing the enumeration. Right now, parcel should have a single status. Therefore, we shouldn't use flags if we don't need to. I'll use a simple enum.

File: Colorblind/Domain/UseCases/ParcelUseCases/ReceiveParcelFromTerminalUseCase.cs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        {
+            throw new DomainError($"Parcel with code {command.Code} doesn't exist!");
+        }
+
+        await _parcelRepository.Update(parcel.Id, command.Version, aggregate =>
+        {
+            if (aggregate.Status != ParcelStatus.Delivered)
+            {
+                throw new DomainError("Parcel was not delivered yet!");
+            }
```
Comment (SimonasKalpokas1) [inline Colorblind/Domain/UseCases/ParcelUseCases/ReceiveParcelFromTerminalUseCase.cs:39]: Don't really understand this part. why check Status in repository? It doesn't look like repository logic and you already have parcel retrieved. And where does the actual update happen (I mean where is the field updated)? For me it looks like the whole await _parcelRepository.Update is unnecessary and just  await _saveChanges.SaveChanges(ct); is enoguh

Comment (edsinkevic) [reply Colorblind/Domain/UseCases/ParcelUseCases/ReceiveParcelFromTerminalUseCase.cs:39]: You are correct, that's not repository logic. We are passing a domain function that will check against an existing parcel in storage. That being said, I wouldn't say that in this case we are checking the status 'in' the repository implementation.

Comment (SimonasKalpokas1) [reply Colorblind/Domain/UseCases/ParcelUseCases/ReceiveParcelFromTerminalUseCase.cs:39]: But you already fetched the parcel from db. Why do another fetch?

Comment (edsinkevic) [reply Colorblind/Domain/UseCases/ParcelUseCases/ReceiveParcelFromTerminalUseCase.cs:39]: The function that we are passing returns an Event basing on the existing parcel in the storage. Then, Marten will see which StreamProjections are reacting to a specific event type. In our case, it's the ParcelProjection. If you look at ParcelProjection.cs file, we describe how some events affect the Parcel aggregate.

Comment (edsinkevic) [reply Colorblind/Domain/UseCases/ParcelUseCases/ReceiveParcelFromTerminalUseCase.cs:39]: You could say it's a Strategy pattern in OOP, the function passing notion. In FP, it's just a function :D

Comment (edsinkevic) [reply Colorblind/Domain/UseCases/ParcelUseCases/ReceiveParcelFromTerminalUseCase.cs:39]: I decided to save some code space and just use a generic Update interface for all kinds of updates.

Comment (edsinkevic) [reply Colorblind/Domain/UseCases/ParcelUseCases/ReceiveParcelFromTerminalUseCase.cs:39]: I know it's super unusual.

Comment (edsinkevic) [reply Colorblind/Domain/UseCases/ParcelUseCases/ReceiveParcelFromTerminalUseCase.cs:39]: I would actually remake it and make sure the API requires the Id. But that will require swiping changes in the front-end. Oof.

**Review by SimonasKalpokas1 — APPROVED** [review]

**Pull request merged**